packages/graphql/src/translate/utils/math.ts

```diff
@@ -0,0 +1,126 @@
+export function matchMathField(graphQLFieldName: string): MathMatch {
+    const mathFieldMatch = graphQLFieldName.match(MATH_FIELD_REGX);
+    if (mathFieldMatch && mathFieldMatch.groups) {
+        const operatorName = mathFieldMatch.groups.operatorName;
+        const propertyName = mathFieldMatch.groups.propertyName;
+        const isMatched = Boolean(mathFieldMatch && mathFieldMatch.length > 2 && operatorName && propertyName);
+        return {
+            isMatched,
+            operatorName,
+            propertyName
+        }
+    } 
+    return {
+        isMatched: false,
+        operatorName: "",
+        propertyName: ""
+    }
+}
+
+export function mathDescriptorBuilder(value: number, entity: GraphElement, graphQLFieldName?: string, fieldMatch?: MathMatch): MathDescriptor {
+    if (!fieldMatch && !graphQLFieldName) {
+        throw new Error('mathDescriptorBuilder need to be invoked with graphQLFieldName or fieldMatch');
+    }
+    let match: MathMatch;
+    if (!fieldMatch) {
+        const mathMatch = matchMathField(graphQLFieldName as string);
+        const {isMatched} = mathMatch;
+
+        if (!isMatched) {
+            throw new Error(`GraphQL field ${graphQLFieldName} is not a valid Math Field`);
+        }
+        match = mathMatch;
+    } else {
+        match = fieldMatch;
+    }
+    const fieldName = match.propertyName;
+    const field = entity.primitiveFields.find((x) => x.fieldName === fieldName);
+    if (!field) {
+        throw new Error(`${fieldName} is not settable`);
+    }
+    return {
+        dbName: mapToDbProperty(entity, fieldName) as string,
+        graphQLType: field.typeMeta.name,
+        fieldName,
+        operationName: match.operatorName,
+        operationSymbol: mathOperatorToSymbol(match.operatorName as string),
+        value
+    };
+}
+
+export function buildMathStatements(mathDescriptor: MathDescriptor, scope: string, param: string): Array<string> {
+    if (mathDescriptor.operationSymbol === "/" && mathDescriptor.value === 0) {
+        throw new Error('Division by zero is not supported');
+    }
+    const statements: string[] = [];
+    statements.push(`WITH *`);
+    statements.push(`CALL {`);
+    statements.push(`WITH ${scope}`);
+    // Raise for operations with NAN
+    statements.push(`CALL apoc.util.validate(apoc.meta.type(${scope}.${mathDescriptor.dbName}) = "NULL", 'Cannot %s %s to Nan', ["${mathDescriptor.operationName}", $${param}])`);
+    const bitSize = mathDescriptor.graphQLType === "Int" ? 32 : 64;
+    // Avoid overflows, for 64 bit overflows, a long overflow is raised anyway by Neo4j
+    statements.push(`CALL apoc.util.validate(${scope}.${mathDescriptor.dbName} ${mathDescriptor.operationSymbol} $${param} > 2^${bitSize-1}-1, 'Value returned from operator %s is larger than %s bit', ["${mathDescriptor.operationName}", "${bitSize}"])`);
+    const cyperType = mathDescriptor.graphQLType === "Int" || mathDescriptor.graphQLType === "BigInt" ? "INTEGER" : "FLOAT";
```

Comment:
Looks like a typo sneaked itself in, `cypherType`

Reply:
That's an incredible eagle-eyed findings :)